src/network/networkd-setlink.c

```diff
@@ -458,7 +458,7 @@ static int link_configure(
                 break;
         }
         case SET_LINK_GROUP:
-                r = sd_netlink_message_append_u32(req, IFLA_GROUP, link->network->group);
+                r = sd_netlink_message_append_u32(req, IFLA_GROUP, (uint32_t)link->network->group);
```

Comment:
Not a matter, but we usually insert a space after the cast: `(uint32_t) link->network->group`.